mixins/labelled-mixin.js

```diff
@@ -73,16 +91,30 @@ export const LabelledMixin = superclass => class extends superclass {
 		};
 	}
 
-	updated(changedProperties) {
+	constructor() {
+		super();
+		this._missingLabelErrorHasBeenThrown = false;
+		this._throwNoLabelExceptionImmediately = false;
+	}
+
+	async updated(changedProperties) {
 		super.updated(changedProperties);
 
+		if (changedProperties.has('label')) {
+			this._validateLabel();
+		}
+
 		if (!changedProperties.has('labelledBy')) return;
 
 		if (this._labelObserver) this._labelObserver.disconnect();
 
 		if (!this.labelledBy) return;
 
+		// try to get the labelling element syncronously first
 		let labelElem = this.getRootNode().querySelector(`#${cssEscape(this.labelledBy)}`);
+		if (!labelElem) {
+			labelElem = await waitForElement(this.getRootNode(), `#${cssEscape(this.labelledBy)}`, 3000);
+		}
```

Comment:
I think this can just be simplified now?

```suggestion
		let labelElem = await waitForElement(this.getRootNode(), `#${cssEscape(this.labelledBy)}`, 3000);
```

And then if `labelElem === null` at this point, I think we can just throw an exception? At this point it has waited 3s and not found it.

Reply:
I was hesitant to do that because by using `waitForElement` I think it will automatically delay by a microtask due to the Promise.  Maybe that's ok to always introduce that delay. I was just being nervous about timing/delays.

I don't think we need to throw a special exception though... since once it waits 3s it will validate after the observer is defined.